PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -0,0 +1,435 @@
+        }
+        // pion <- lambda
+        hTrackablePtEta->Fill(kTrackableAll, pionFromLambda.pt(), pionFromLambda.eta());
+        if (hasITS[pionFromLambda.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableITS, pionFromLambda.pt(), pionFromLambda.eta());
+        }
+        if (hasTPC[pionFromLambda.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableTPC, pionFromLambda.pt(), pionFromLambda.eta());
+        }
+        if (hasITS[pionFromLambda.globalIndex()] && hasTPC[pionFromLambda.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableITSTPC, pionFromLambda.pt(), pionFromLambda.eta());
+        }
+        // proton <- lambda
+        hTrackablePtEta->Fill(kTrackableAll, proton.pt(), proton.eta());
+        if (hasITS[proton.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableITS, proton.pt(), proton.eta());
+        }
+        if (hasTPC[proton.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableTPC, proton.pt(), proton.eta());
+        }
+        if (hasITS[proton.globalIndex()] && hasTPC[proton.globalIndex()]) {
+          hTrackablePtEta->Fill(kTrackableITSTPC, proton.pt(), proton.eta());
+        }
+      } // close tracked if
+
+      // with pion track cuts (see checkTrackGlbTrk and checkTrkItsTrk)
+      if (selectedIts[pionId]) {
+        hCandidates->Fill(kHFStepItsTrackableCuts, mcParticle.pt(), mass, true, mcParticle.originGen());
+        if (!inAcceptance) {
+          LOGP(info, "Candidate {} has daughters not in acceptance but pion <-- charm tracked and selected (its only)", mcParticle.globalIndex());
+        }
+      }
+      if (selectedItsTpc[pionId]) {
+        hCandidates->Fill(kHFStepItsTpcTrackableCuts, mcParticle.pt(), mass, true, mcParticle.originGen());
+        if (!inAcceptance) {
+          LOGP(info, "Candidate {} has daughters not in acceptance but pion <-- charm tracked and selected (its & tpc)", mcParticle.globalIndex());
+        }
+      }
+
+    } // close loop mcParticles
+  }   // close candidateMcLoop
+
+  // process functions
+  void processXic0(candidateInfo const& candidates,
+                   genLevelInfo const& genParticles,
+                   TracksWithSelectionMC const& tracks,
+                   aod::McCollisionLabels const& colls)
+  {
+    int pdgCode = Pdg::kXiC0;
+    candidateFullLoop(candidates, genParticles, tracks, colls, pdgCode);
+  }
+  PROCESS_SWITCH(HfTaskMcEfficiencyToXiPi, processXic0, "Process MC for Xic0 signal", true);
+
+  void processOmegac0(candidateInfo const& candidates,
+                      genLevelInfo const& genParticles,
+                      TracksWithSelectionMC const& tracks,
+                      aod::McCollisionLabels const& colls)
+  {
+    int pdgCode = Pdg::kOmegaC0;
+    candidateFullLoop(candidates, genParticles, tracks, colls, pdgCode);
+  }
+  PROCESS_SWITCH(HfTaskMcEfficiencyToXiPi, processOmegac0, "Process MC for Omegac0 signal", false);
```

Comment:
1. Why can you not have a single process function?
2. Do you want to allow running both `processXic0` and `processOmegac0` together?

Reply:
I do not want to run processXic0 and processOmegac0 together, I added a check in the init function

Reply:
The only difference between the two process functions is the value of `pdgCode` which you can control with a configurable. I don't see any benefit of having two process functions and the `candidateFullLoop` function.

Reply:
1) I have implemented different process functions because I followed "the style" of the other existing task for MC efficiency calculation (a process function for each set of particles to take into account)

2) I would like to have two separate loops because the first one is based only the reconstructed information, the second on the first one + on generated information. To better highlight this difference I changed the name of the first funtion

Reply:
1. In the other efficiency task, the process functions are needed because of different combinations of tables they consume. Your task consumes the same tables in both cases, so splitting it into two process functions is just a code duplication.
2. I am not discouraging your from having `candidateRecLoop`. I was talking about `candidateFullLoop` which is called once in each `process` function with the same parameters, making it pointless.

Reply:
Okay, I switched to 1 process function only

Reply:
Perfect! Thanks!